client/go/vespa/application.go

```diff
@@ -146,6 +146,55 @@ func (ap *ApplicationPackage) zipReader(test bool) (io.ReadCloser, error) {
 	return f, nil
 }
 
+func (ap *ApplicationPackage) Unzip(test bool) (string, error) {
+	if !ap.IsZip() {
+		return "", fmt.Errorf("can't unzip a package that is a directory structure")
+	}
+	tmp, err := os.MkdirTemp(os.TempDir(), "vespa-test-pkg")
+	if err != nil {
+		return "", err
+	}
+	path := ap.Path
+	if test {
+		path = ap.TestPath
+	}
+	f, err := zip.OpenReader(path)
```

Comment:
If this fails, `tmp` will still exist. How about this:
```
cleanTemp := true
tmp, err := os.MkdirTemp(os.TempDir(), "vespa-test-pkg")
if err != nil {
    return "", err
}
defer func() {
    if cleanTemp {
        os.RemoveAll(tmp)
    }
}()
```

then remove the explicit `os.RemoveAll`s below and simply do `cleanTemp = false` before returning.

Reply:
I like it.